backend/routers/grafana.py

```
@@ -0,0 +1,318 @@
+    if not user or not api_key:
+        return {}
+    encoded = base64.b64encode(f"{user}:{api_key}".encode()).decode()
+    return {"Authorization": f"Basic {encoded}"}
+
+
+async def _grafana_get(url: str, params: dict, user: str, api_key: str) -> JSONResponse:
+    async with httpx.AsyncClient(timeout=15.0) as client:
+        resp = await client.get(url, params=params, headers=_auth_headers(user, api_key))
+    try:
+        body = resp.json()
+    except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)